src/NServiceBus.Core/Timeout/Hosting/Windows/TimeoutDispatcherProcessor.cs

```diff
@@ -19,20 +21,47 @@ public TimeoutDispatcherProcessor()
 
         public TimeoutPersisterReceiver TimeoutPersisterReceiver { get; set; }
 
+        public PipelineExecutor PipelineExecutor { get; set; }
+
         public Configure Configure { get; set; }
       
         public Address InputAddress { get; set; }
 
+        public ReadOnlySettings Settings { get; set; }
+
         public bool Disabled { get; set; }
 
         public bool Handle(TransportMessage message)
         {
             var timeoutId = message.Headers["Timeout.Id"];
-            TimeoutData timeoutData;
 
-            if (TimeoutsPersister.TryRemove(timeoutId, out timeoutData))
+            var persisterV2 = TimeoutsPersister as IPersistTimeoutsV2;
+            if (persisterV2 != null)
+            {
+                var timeoutData = persisterV2.Peek(timeoutId);
+                if (timeoutData == null)
+                {
+                    return true;
+                }
+
+                var sendOptions = timeoutData.ToSendOptions(Configure.LocalAddress);
+                
+                if (ShouldSuppressTransaction())
+                {
+                    sendOptions.EnlistInReceiveTransaction = false;
```

Comment:
I checked both, MSMQ only suppresses native transactions while SQL suppresses all transactions as far as I understood the code. But probably better if a second pair of eyes verifies this.